tsl/src/compression/api.c

```diff
@@ -1088,9 +1088,17 @@ recompress_chunk_segmentwise_impl(Chunk *uncompressed_chunk)
 	 * on the updated tuple in the CHUNK table potentially preventing other transaction
 	 * from updating it
 	 */
-	ts_chunk_clear_status(uncompressed_chunk,
-						  CHUNK_STATUS_COMPRESSED_UNORDERED | CHUNK_STATUS_COMPRESSED_PARTIAL);
+	if (ts_chunk_clear_status(uncompressed_chunk,
+							  CHUNK_STATUS_COMPRESSED_UNORDERED | CHUNK_STATUS_COMPRESSED_PARTIAL))
+		ereport(LOG,
+				(errmsg("cleared chunk status for recompression: \"%s.%s\"",
+						NameStr(uncompressed_chunk->fd.schema_name),
+						NameStr(uncompressed_chunk->fd.table_name))));
 
+	ereport(LOG,
+			(errmsg("acquiring locks for recompression: \"%s.%s\"",
+					NameStr(uncompressed_chunk->fd.schema_name),
+					NameStr(uncompressed_chunk->fd.table_name))));
```

Comment:
Might be good to add the lock names as well so that you do not have to jump back and forth between the code and the log when debugging.

Reply:
+1